File: task3.js
const fs = require('fs');
const readline = require('readline');

function proverka(x) {
    let sum = 0, proizv = 1;
    let n = Math.abs(x);
    while (n > 0) {
        let d = n % 10;
        sum += d;
        proizv *= d;
        n = Math.floor(n / 10);
    }
    return proizv > sum;
}

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    let lines = [];
    rl.on('line', line => lines.push(line.trim()));
    rl.on('close', () => {
        const N = parseInt(lines[0]);
        const numbers = lines.slice(1, 1 + N).map(Number);
        const ans = numbers.filter(x => proverka(x));
        if (ans.length === 0) {
            console.log("не найдено");
        } else {
            console.log(ans.join(' '));
        }
    });

    /*
    const input = fs.readFileSync('input.txt', 'utf8').trim().split(/\s+/);
    const N = parseInt(input[0]);
    const numbers = input.slice(1, 1 + N).map(Number);
    const ans = numbers.filter(x => proverka(x));
    const output = ans.length === 0 ? 'не найдено' : ans.join(' ');
    fs.writeFileSync('output.txt', output + '\n');
    */
}

main();

File: task3.rs
use std::io::{self, Write};

fn proverka(mut x: u64) -> bool {
    let mut sum = 0;
    let mut proizv = 1;
    while x > 0 {
        let d = x % 10;
        sum += d;
        proizv *= d;
        x /= 10;
    }
    proizv > sum
}

fn main() {
    let mut input = String::new();
    print!("введите N: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut ans = Vec::new();
    println!("введите N чисел:", n);
    for _ in 0..n {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let x: u64 = input.trim().parse().unwrap();
        if proverka(x) {
            ans.push(x);
        }
    }

    if ans.is_empty() {
        println!("не найдено");
    } else {
        for x in ans {
            print!("{} ", x);
        }
        println!();
    }

    /*
    use std::fs::File;
    use std::io::BufRead;
    let fin = File::open("input.txt").unwrap();
    let reader = io::BufReader::new(fin);
    let mut lines = reader.lines();
    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let mut ans = Vec::new();
    for _ in 0..n {
        let x: u64 = lines.next().unwrap().unwrap().trim().parse().unwrap();
        if proverka(x) {
            ans.push(x);
        }
    }
    let mut fout = File::create("output.txt").unwrap();
    if ans.is_empty() {
        writeln!(fout, "не найдено").unwrap();
    } else {
        for x in ans {
            write!(fout, "{} ", x).unwrap();
        }
        writeln!(fout).unwrap();
    }
    */
}
